src/Misc/baseError.ts

```diff
@@ -1,12 +1,46 @@
+import { ErrorCodesType } from "./errorCodes";
+
 /**
+ * BabylonJS error internal class wrapper. Extending Error for a class with docs
+ * fails typedoc validation, because lib.es5.d.ts does not have any docs for Error.
  * @ignore
- * Application error to support additional information when loading a file
  */
-export abstract class BaseError extends Error {
+class BabylonJSErrorWrapper extends Error {}
+
+/**
+ * Base Error class for babylonjs errors
+ */
+export class BaseError extends BabylonJSErrorWrapper {

```

Comment:
I find it odd that we are exposing/using BaseError for errors. This class is intended to be an abstract base class.